src/Blammo/Logging/Simple.hs

```diff
@@ -16,6 +17,13 @@ import Control.Monad.IO.Unlift (MonadUnliftIO)
 newLoggerEnv :: MonadIO m => m Logger
 newLoggerEnv = liftIO $ newLogger =<< Env.parse
 
+-- | Initialize logging (configured via environment variables),
+--   pass a 'Logger' to the callback, and clean up at the end
+--
+-- Applications should avoid calling this more than once in their lifecycle.
+withLoggerEnv :: MonadUnliftIO m => (Logger -> m a) -> m a
```

Comment:
I was a little confused why you had to deriving `MonadUnliftIO` on `WithLogger` and this is why. I think this only need be `MonadIO`, right?

Reply:
It's to do stuff like `appSqlPool <- runWithLogger appLogger makePostgresPool` because `makePostgresPool` has a `MonadUnliftIO` constraint.

Reply:
(The two changes are unrelated, they're just both things I happened to find useful while integrating into megarepo/backend.)

Reply:
I see. So if you intend to allow the newtype to be usable directly, not only through deriving-via, then it will need whatever reasonable instances we can think of. I would not stop at `MonadUnliftIO` then; you'll want state, error/except, probably mask/catch/throw right? Maybe even `MonadValidate`? I know we can add them as we hit them, but I might start with whatever we can do without any extra dependencies here (which is a lot) -- what do you think?

This is orthogonal though, I think `withLoggerEnv` still only needs `MonadIO` right?

Reply:
I don't want to be in a situation where we're following ad hoc desires in throwing any any all instances on here, but `MonadUnliftIO` seems like a pretty fair one.

Reply:
On `withLoggerEnv`, `MonadUnliftIO` is needed because `withLogger` uses it for the unliftio version of `finally`.

Reply:
> `finally`

That's what I missing, sorry.